lambda-runtime/src/AWS/Lambda/Client.hs

```diff
@@ -0,0 +1,192 @@
+module AWS.Lambda.Client
+  ( HTTPConfig
+  , LambdaClient
+  , InternalLambdaClientError (..)
+  , LambdaResponse (..)
+  , getHttpConfig
+  , getNextLambdaResponse
+  , sendEventError
+  , sendEventSuccess
+  , sendBootError
+  )
+where
+
+import AWS.Prelude
+import Control.Exception (displayException)
+import Control.Monad.Except (ExceptT(..), liftEither, throwError, withExceptT)
+import System.Environment (lookupEnv)
+
+import qualified Data.Aeson          as JSON
+import qualified Data.ByteString     as BS
+import qualified Network.HTTP.Client as HTTP
+import qualified Network.HTTP.Types  as HTTP
+
+data HTTPConfig = HTTPConfig
+  { request :: HTTP.Request
+  , manager :: HTTP.Manager
+  }
+
+data LambdaResponse = LambdaResponse
+  { event     :: JSON.Value
+  , requestId :: RequestId
+  }
+  deriving stock Show
+
+newtype RequestId = RequestId Text
+  deriving stock (Show)
+
+instance Conversion BS.ByteString RequestId where
+  convert (RequestId requestId) = encodeUtf8 requestId
+
+data InternalLambdaClientError
+  = LambdaContextDecodeError Text
+  | MissingLambdaRunTimeApi
+  | InvalidLambdaRunTimeApi Text
+  | ConnectionError HTTP.HttpException
+  | ResponseBodyDecodeFailure Text
+  | UnExpectedRunTimeError HTTP.Status Text
+  | PayLoadTooLarge Text
```

Comment:
@epicallan Same here, I'd probably have `Payload` instead.

Reply:
This error constructor is actually specifically used when the payload is too large. We know that from the 413 HTTP status code returned in an event. Naming it `Payload` may imply it can be used for other cases which is not the case.

Reply:
@epicallan Oh, I think he meant that "Payload" is a single word so this should probably be `PayloadTooLarge`.

Reply:
Ok, I get it. Thanks for the clarrification.